python/GafferTest/FileSystemPathTest.py

```diff
@@ -304,6 +304,40 @@ def testMixedSlashes( self ) :
 			self.assertEqual( list( p ), [ r"unc\tshare", r"must\twork" ] )
 			self.assertEqual( p.nativeString(), r"/unc\tshare/must\twork")
 
+	def testEscapedCharacters( self ) :
+
+		p = Gaffer.FileSystemPath( r"/\~this/\#must/\$be/escaped.txt" )
+		if os.name == "nt" :
+			self.assertEqual( list( p ), [ r"\#must", r"\$be", "escaped.txt" ] )
+			self.assertEqual( p.root(), r"//\~this/")
+			self.assertEqual( str( p ), r"//\~this/\#must/\$be/escaped.txt" )
+			self.assertEqual( p.nativeString(), r"\\\~this\\#must\\$be\escaped.txt" )
```

Comment:
Might be good to have another test that asserts that `p2 = Gaffer.FileSystemPath( p.nativeString() )` gets us an identical path to `p`?